keyspace/key_space.go

```diff
@@ -334,6 +364,14 @@ func (ks *KeySpace) onUpdate() {
 // expected to be greater than or equal to the current one; otherwise, it
 // should be strictly greater.
 func patchHeader(h *etcdserverpb.ResponseHeader, update etcdserverpb.ResponseHeader, allowSameRevision bool) error {
+	if err := checkHeader(h, update, allowSameRevision); err != nil {
```

Comment:
nit: `patchHeader` has just one remaining usage. I think the indirection is no longer serving us and it should be inlined to the remaining call site.